src/utils.ts

```diff
@@ -165,6 +166,14 @@ export function parseUsername(input: string): string {
   return parts[parts.length - 1];
 }
 
+export function parseProjectURL(pathname: string): { content: ProjectContent; commit: string; path: string } {
```

Comment:
Ok this could be a interesting discussion point.. Trying to obtain information from the nested routes inside the `View` component, the easiest and cleanest solution I found was to parse the pathname starting before `tree` or `history` and parse the commit and path that one passes after that, and in case of missing parts to provide fallbacks..
A URL without `tree` or `history` does not exist since the page redirects either to `history` or `tree` as fallback.